relayer/src/chain/cosmos.rs

```diff
@@ -276,9 +276,25 @@ impl CosmosSdkChain {
 
         let response = self.block_on(broadcast_tx_sync(self, tx_bytes))?;
 
-        debug!("[{}] send_tx: broadcast_tx_sync: {:?}", self.id(), response);
-
-        self.incr_account_sequence()?;
+        match response.code {
+            tendermint::abci::Code::Ok => {
+                // A success means the account s.n. was increased
+                self.incr_account_sequence()?;
+                debug!("[{}] send_tx: broadcast_tx_sync: {:?}", self.id(), response);
+            }
+            tendermint::abci::Code::Err(_) => {
+                // Avoid increasing the account s.n. if CheckTx failed
+                // Log the error
+                error!("[{}] send_tx: broadcast_tx_sync: {:?}", self.id(), response);
+                // The primary reason (we know of) causing broadcast_tx_sync to fail
+                // is due to "out of gas" errors. These are unrecoverable at the moment
+                // on the Hermes side. We'll inform the user to check for misconfig.
+                error!(
```

Comment:
Should we check for the error code 11 to display the out of gas error message? https://github.com/cosmos/cosmos-sdk/blob/v0.42.10/types/errors/errors.go#L50

Reply:
This is a very good point I overlooked, thanks Soares, fixing now!